src/pages/index.js

```diff
@@ -0,0 +1,312 @@
+    })
+    .catch((err) => console.error(`Error: ${err}`))
+    .finally(() => {
+      submitButton.textContent = "Save";
+    });
+}
+
+// Handle new post form
+function handleNewPostFormSubmit(evt) {
+  evt.preventDefault();
+  const newPostModal = document.querySelector("#new__post-modal");
+  const submitButton = newPostModal.querySelector(".modal__submit-btn");
+  submitButton.textContent = "Saving...";
+  const link = newPostModalLinkInput.value.trim();
+  const name = newPostModalCaptionInput.value.trim();
+  const newCard = { name, link };
+
+  api
+    .addCard(newCard)
+    .then((card) => {
+      const cardElement = getCardElement(card, userId);
+      cardsList.prepend(cardElement);
+      closeModal(newPostModal);
+      newPostFormElement.reset();
+      disableButton(newPostSubmitbtn, validationConfig);
+    })
+    .catch((err) => {
+      console.error("Error adding card:", err);
+    })
+    .finally(() => {
+      submitButton.textContent = "Save";
+    });
+}
+
+// Open avatar modal
+avatarEditBtn.addEventListener("click", () => {
+  openModal(avatarModal);
+});
+
+// Handle avatar form submission
+avatarForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+
+  const submitButton = avatarForm.querySelector(".modal__submit-btn");
+  submitButton.textContent = "Saving...";
+
+  api
+    .setUserAvatar({ avatar: avatarInput.value })
+    .then((userData) => {
+      profileAvatar.src = userData.avatar;
+      closeModal(avatarModal);
+      avatarForm.reset();
+      disableButton(submitButton, validationConfig);
+    })
+    .catch((err) => console.error(`Error: ${err}`))
+    .finally(() => {
+      submitButton.textContent = "Save"; // Reset button text
+    });
+});
+
+// Open profile edit modal
+profileEditButton.addEventListener("click", () => {
+  profileEditModalNameInput.value = profileName.textContent;
+  profileEditModalDescriptionInput.value = profileDescription.textContent;
+  resetFormValidation(profileEditFormElement, validationConfig);
+  openModal(profileEditModal);
+});
+
+// Open new post modal
+newPostButton.addEventListener("click", () => {
+  openModal(newPostModal);
+});
+
+// Submit handlers
+profileEditFormElement.addEventListener("submit", handleProfileFormSubmit);
+newPostFormElement.addEventListener("submit", handleNewPostFormSubmit);
+
+// Initialize API
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d281195a-2927-4998-9155-1ddf6322a50b",
+    "Content-Type": "application/json",
+  },
+});
+
+// Load user and card data
+api.getAppInfo().then(([cards, userInfo]) => {
+  const userId = userInfo._id;
+  cards.forEach((card) => {
+    const cardElement = getCardElement(card, userId);
+    cardsList.append(cardElement);
+  });
+
+  profileName.textContent = userInfo.name;
+  profileDescription.textContent = userInfo.about;
+  profileAvatar.src = userInfo.avatar;
+});
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**